src/builderGenerator.ts

```diff
@@ -274,7 +276,7 @@ function addBuilderMethodToImports({
     const valueFunctionParameterType = `((builder: ${builderType}) => ${builderType})`;
     return {
       name: `${valueParameterTypeName} | ${valueFunctionParameterType}`,
-      statement: `this.obj.${propName} = typeof val === 'function' ? val(${builderMethodName}()).get() : val;`,
+      statement: `this.obj.${propName} = typeof val === 'function' ? val(${builderMethodName}(this.obj.${propName})).get() : val;`,
```

Comment:
if i understand correctly the way to use this new feature is if user does something like this:
```typescript
aSomeBuilder()
  .withProp1({...}) // could be scalar/plain object/builder
  .withProp1(builderWithPrevProp1 => builderWithPrevProp1.withProp2(5)) // builderWithPrevProp1 has prop1 from previous line
  .get()
```

IMO this leads to messier code cause there's "magic" code from previous builder config that you need to dig through in order to understand what's going on. also, user expects that a call to `prop1` will override previous config and not "build on top of it".
If we want this behavior i prefer to set it explicitly, something like this:
```typescript
`((builder: ${builderType}, shouldBuildFromCurrentProperty: boolean = false) => ${builderType})`;
    return {
      name: `${valueParameterTypeName} | ${valueFunctionParameterType}`,
      statement: `this.obj.${propName} = typeof val === 'function' ? val(${builderMethodName}()).get() : val;`,
      statement: `this.obj.${propName} = typeof val === 'function' ? val(${builderMethodName}(shouldBuildFromCurrentProperty ? this.obj.${propName} : undefined)).get() : val;`,
```